tests/main_tests.nim

```diff
@@ -169,16 +169,103 @@ suite "summary":
     check mySummary.valueByName("mySummary_count") == 2
     check mySummary.valueByName("mySummary_sum") == 10.5
 
+  test "alternative API":
+    summary("one_off_summary").observe(10)
+    check summary("one_off_summary").valueByName("one_off_summary_count") == 1
+    check summary("one_off_summary").valueByName("one_off_summary_sum") == 10
+
   test "timing":
     mySummary.time:
       sleep(1000)
-      check(mySummary.valueByName("mySummary_sum") == 0)
-    check(mySummary.valueByName("mySummary_sum") >= 1)
+      check mySummary.valueByName("mySummary_sum") == 0
+    check mySummary.valueByName("mySummary_sum") >= 1
 
   test "timing with labels":
     declareSummary lsummary, "help", ["foobar"], registry = registry
     let labelValues = ["b"]
     lsummary.time(labelValues):
       sleep(1000)
-    check(lsummary.valueByName("lsummary_sum", labelValues) >= 1)
+    check lsummary.valueByName("lsummary_sum", labelValues) >= 1
+
+suite "histogram":
+  setup:
+    var registry = newRegistry()
+    declareHistogram myHistogram, "help", registry = registry
+
+  test "basic":
+    check myHistogram.valueByName("myHistogram_bucket", [], ["1.0"]) == 0
```

Comment:
What's the rationale for having these stingy APIs instead of using regular object fields?

Reply:
Metric objects having different names from Collector objects, in this case.